latex_access/tests/test_ueb.py

```diff
@@ -0,0 +1,160 @@
+        self.braille.remove_dollars = False
+        self.assertEqual(self.braille.uebDollar('%', 0), ('`s', 0))
+        self.braille.remove_dollars = True
+        self.assertEqual(self.braille.uebDollar('%', 0), ('', 0))
+
+    def test_numbers(self):
+        """Tests translations of numbers."""
+        self.braille.lastnumber = -1
+        self.assertEqual(self.braille.numbers('0', 0), ('#j', 0))
+        self.braille.lastnumber = 1
+        self.assertEqual(self.braille.numbers('{11}', 2), ('a', 2))
+
+    def test_dot_in_the_middle_of_a_number(self):
+        """Tests translations of dots between numbers."""
+        # Simulate a case when dot is directly after a digit.
+        self.braille.lastnumber = 0 
+        self.assertEqual(self.braille.dot('1', 1), ('4', 1))
+        # Ensure that last number is set to the starting position.
+        self.assertEqual(self.braille.lastnumber, 1)
+
+    def test_dot(self):
+        """Tests translations of dots."""
+        # We need to call `before` manually,
+        # to set position of the last encountered number.
+        # Normally this is taken care of by the base `translate` method.
+        self.braille.before()
+        self.assertEqual(self.braille.dot('1', 1), ('4', 1))
+        # Ensure that start number remains unchanged.
+        self.assertEqual(self.braille.lastnumber, -1)
+
+    def test_comma_between_digits(self):
+        """Tests translations of commas when they are between digits."""
+        # Simulate a case when comma is directly after a digit.
+        self.braille.lastnumber = 0
+        self.assertEqual(self.braille.comma('1', 1), ('1', 1))
+        # Ensure that last number is set to the starting position.
+        self.assertEqual(self.braille.lastnumber, 1)
+
+    def test_comma(self):
+        """Tests translations of commas."""
+        # We need to call `before` manually,
+        # to set position of the last encountered number.
+        # Normally this is taken care of by the base `translate` method.
+        self.braille.before()
+        self.assertEqual(self.braille.comma('1', 1), ('1', 1))
+        # Ensure that start number remains unchanged.
+        self.assertEqual(self.braille.lastnumber, -1)
+
+    def test_letterSign(self):
+        """Tests translations of letter sign."""
+        self.braille.lastnumber = 0
+        self.assertFalse(self.braille.letterSign('A', 0))
+        self.assertTrue(self.braille.letterSign('B', 0))
+        self.braille.lastnumber = 1
+        self.assertTrue(self.braille.letterSign('{ B }', 2))
+        self.assertTrue(self.braille.letterSign('{ }', 0))
+        self.assertTrue(self.braille.letterSign('{,}', 2))
+        self.assertTrue(self.braille.letterSign('{b,}', 1))
+        self.assertTrue(self.braille.letterSign('{,b }', 2))
+        self.assertTrue(self.braille.letterSign(' b', 1))
+
+    def test_lowerLetter(self):
+        """Tests translations of lower letters."""
+        self.braille.lastnumber = None
```

Comment:
In the code `lastnumber` is always set to a digit, and never to `None`, Either change this to a number, or add an explanatory comment, which clarifies why `None` is appropriate here.